lib/ansible/galaxy/api.py

```diff
@@ -322,7 +341,7 @@ def _call_galaxy(self, url, args=None, headers=None, method=None, auth_required=
                 expires = datetime.datetime.strptime(server_cache[url_info.path]['expires'], iso_datetime_format)
                 valid = datetime.datetime.utcnow() < expires
 
-            if valid and not url_info.query:
+            if valid and not ('page' in query or 'offset' in query):
```

Comment:
The second condition became complex enough that id turn it into a named expression:
```suggestion
            is_paginated_url = 'page' in query or 'offset' in query
            if valid and not is_paginated_url:
```
WDYT?